### DISCUSSION TITLE
Support of Gremlin sessions

### DISCUSSION THREAD

**Discussion by yongkansage** [Gremlinq/ExRam.Gremlinq#227] [Q&A]
Does the library support Gremlin sessions (https://docs.aws.amazon.com/neptune/latest/userguide/access-graph-gremlin-sessions.html)?

**Comment by danielcweber** [comment]
The lacking support for sessions is due to the reasoning I described [here](https://github.com/ExRam/ExRam.Gremlinq/discussions/225#discussioncomment-573825). Provider support for explicit transactions/sessions is just not really a thing (at least not in any provider that I am aware of). Plus, sessions are not supported on the bytecode level as described [here](https://stackoverflow.com/a/65217991/1714634).

**Reply by yongkansage** [reply]
Given an example, say, Vertex A and a collection of Vertex B's need to be added to the graph, along with relationships between A and each B's.  All these need be done as one atomic transaction.  Do you think this is something for text/script query with Gremlin session?

**Reply by danielcweber** [reply]
Sure, but as I said, transactions today are merely implicit (for Neptune anyway, CosmosDB doesn't support them at all). So at this point, explicit session support in Gremlinq is rather pointless.

**Comment by tatish86** [comment]
Hello,
I see that support was added for transactions in CosmosDB - https://docs.microsoft.com/en-us/azure/cosmos-db/sql/transactional-batch.
Does this mean you will rethink about adding transaction support?

**Reply by jbockle** [reply]
thats for sql/core api, transactions are still not supported by gremlin api - https://docs.microsoft.com/en-us/azure/cosmos-db/graph/gremlin-support#unsupported-features
